### ISSUE TITLE
ipwrap filter doesn't work with generators

### ISSUE THREAD

**Issue by hfinucane** [ansible/ansible#11434]
Example:

```
  - debug: msg="{{ ['fe80::a236:9fff:fe5f:6af4']|ipwrap|join(',') }}"
  - debug: msg="{{ ['fe80::a236:9fff:fe5f:6af4']|map('string')|ipwrap|join(',') }}"
```

`string` being a filter plugin (http://jinja.pocoo.org/docs/dev/templates/#string) that, confusingly, does what `unicode()` does in Python. It returns a generator.

I would expect the output to be the same, but actually:

```
TASK [debug msg=[fe80::a236:9fff:fe5f:6af4]] ************************************
ok: [xxx] => {
    "msg": "[fe80::a236:9fff:fe5f:6af4]", 
    "changed": false
}

TASK [debug msg=fe80::a236:9fff:fe5f:6af4] **************************************
ok: [xxx] => {
    "msg": "fe80::a236:9fff:fe5f:6af4", 
    "changed": false
}

```

This is true in 1.9.1 and in devel.

**Comment by sivel** [comment]
Can you please fill out the [issue template](https://github.com/ansible/ansible/blob/devel/ISSUE_TEMPLATE.md) and provide it in this ticket so that we have more information including how to reproduce your error?

**Comment by hfinucane** [comment]
(yeah, I hit enter before entering anything- sorry)

**Comment by hfinucane** [comment]
Actually, I think I'm wrong- the root cause isn't that they're unicode, it's that the various Jinja2 filters pass _generators_ around, and there's a bunch of hardcoded tests for 'is this a list or a tuple'.

**Comment by hfinucane** [comment]
Additionally, now that I've figured this out, an effective workaround is piping things through the `list` filter before sending them to `ipwrap`.

**Comment by bcoca** [comment]
closing via merged #11435
